base_business_document_import/models/business_document_import.py

```diff
@@ -177,20 +187,21 @@ def _match_partner(self, partner_dict, chatter_msg, partner_type="supplier"):
                     % (partner_type_label, partner_domain, partner_type_label)
                 )
                 return partner
-        if partner_dict.get("ref"):
-            partner = rpo.search(
-                domain + [("ref", "=", partner_dict["ref"])], limit=1, order=order
-            )
-            if partner:
-                return partner
-        if partner_dict.get("name"):
-            partner = rpo.search(
-                domain + [("name", "=ilike", partner_dict["name"])],
-                limit=1,
-                order=order,
-            )
+
+        if partner_dict.get("vat"):
```

Comment:
@jbaudoux You already make a check on vat a line #123. Why do you move the search on vat here with a domain that could contain others criteria?

Reply:
@lmignon If I cannot match the right partner inside the commercial entity structure, then I match the one with the highest customer/supplier rank